File: Serie/src/main/java/com/ferrazeduardo/escola/Serie/domain/Serie.java
package com.ferrazeduardo.escola.Serie.domain;
import java.math.BigDecimal;
import java.util.Date;
import java.util.List;
import java.util.UUID;


public class Serie {


    public Serie(int AA_MATRICULA, int QT_AVALIACAO, UUID ID_UNIDADE, BigDecimal VL_MEDIA, Date DT_INICIO, Date DT_FIM) {
        this.AA_MATRICULA = AA_MATRICULA;
        this.QT_AVALIACAO = QT_AVALIACAO;
        this.ID_UNIDADE = ID_UNIDADE;
        this.VL_MEDIA = VL_MEDIA;
        this.DT_INICIO = DT_INICIO;
        this.DT_FIM = DT_FIM;
    }


    public Serie(UUID id,int AA_MATRICULA, int QT_AVALIACAO, UUID ID_UNIDADE, BigDecimal VL_MEDIA, Date DT_INICIO, Date DT_FIM) {
        this.AA_MATRICULA = AA_MATRICULA;
        this.QT_AVALIACAO = QT_AVALIACAO;
        this.ID_UNIDADE = ID_UNIDADE;
        this.VL_MEDIA = VL_MEDIA;
        this.DT_INICIO = DT_INICIO;
        this.DT_FIM = DT_FIM;
        this.id = id;
    }

    private UUID id;

    public UUID getId() {
        return id;
    }

    public int getAA_MATRICULA() {
        return AA_MATRICULA;
    }

    public int getQT_AVALIACAO() {
        return QT_AVALIACAO;
    }

    public UUID getID_UNIDADE() {
        return ID_UNIDADE;
    }

    public BigDecimal getVL_MEDIA() {
        return VL_MEDIA;
    }

    public Date getDT_INICIO() {
        return DT_INICIO;
    }

    public Date getDT_FIM() {
        return DT_FIM;
    }

    private int AA_MATRICULA;
    private int QT_AVALIACAO;
    private UUID ID_UNIDADE;
    private BigDecimal VL_MEDIA;
    private Date DT_INICIO;
    private Date DT_FIM;

    private List<ValorSerie> valores;

    public Unidade getUnidade() {
        return unidade;
    }

    public void setUnidade(Unidade unidade) {
        this.unidade = unidade;
    }

    public List<Materia> getMaterias() {
        return materias;
    }

    public void setMaterias(List<Materia> materias) {
        this.materias = materias;
    }

    public List<ValorSerie> getValores() {
        return valores;
    }

    public void setValores(List<ValorSerie> valores) {
        this.valores = valores;
    }

    private List<Materia> materias;
    private Unidade unidade;
}

File: Serie/Serie.Domain/Entity/Serie.cs
using System;

namespace Serie.Domain.Entity;

public class Serie : SeedWorks.Entity
{

    public Serie(int AA_MATRICULA, int QT_AVALIACAO, Rede rede, Decimal VL_MEDIA, DateTime DT_INICIO, DateTime DT_FIM) {
        this.AA_MATRICULA = AA_MATRICULA;
        this.QT_AVALIACAO = QT_AVALIACAO;
        this.rede = rede;
        this.VL_MEDIA = VL_MEDIA;
        this.DT_INICIO = DT_INICIO;
        this.DT_FIM = DT_FIM;
    }


    public Serie(Guid id,int AA_MATRICULA, int QT_AVALIACAO, Rede rede, Decimal VL_MEDIA, DateTime DT_INICIO, DateTime DT_FIM) {
        this.AA_MATRICULA = AA_MATRICULA;
        this.QT_AVALIACAO = QT_AVALIACAO;
        this.rede = rede;
        this.VL_MEDIA = VL_MEDIA;
        this.DT_INICIO = DT_INICIO;
        this.DT_FIM = DT_FIM;
        this.setId(id);
    }


    public int getNr_sala() {
        return nr_sala;
    }

    public void setNr_sala(int nr_sala) {
        this.nr_sala = nr_sala;
    }

    private int nr_sala;

  

    public int getAA_MATRICULA() {
        return AA_MATRICULA;
    }

    public int getQT_AVALIACAO() {
        return QT_AVALIACAO;
    }


    public Decimal getVL_MEDIA() {
        return VL_MEDIA;
    }

    public DateTime getDT_INICIO() {
        return DT_INICIO;
    }

    public DateTime getDT_FIM() {
        return DT_FIM;
    }

    public int AA_MATRICULA { get; private set; }
    public int QT_AVALIACAO { get; private set; }
    public Decimal VL_MEDIA { get; private set; }
    public DateTime DT_INICIO { get; private set; }
    public DateTime DT_FIM { get; private set; }


    public Rede getRede() {
        return rede;
    }

    public void setRede(Rede rede) {
        this.rede = rede;
    }


    public void setMaterias(List<Materia> materias) {
        this.materias = materias;
    }

  

   

    public ICollection<Materia> materias { get; private set; } = [];
    public Rede rede { get; private set; }
    public Sala sala { get; private set; }
    public ICollection<ValorSerie> ValorSeries { get; private set; } = [];
    public ICollection<Aula> aulas { get; private set; } = [];

  
}
